soul-plugin/soul-plugin-apache-dubbo/src/test/java/org/dromara/soul/plugin/apache/dubbo/ApacheDubboPluginTest.java

```diff
@@ -62,18 +70,58 @@ public void setUp() {
         metaData.setMethodName("findAll");
         metaData.setRpcType(RpcTypeEnum.DUBBO.getName());
         ApacheDubboProxyService apacheDubboProxyService = mock(ApacheDubboProxyService.class);
-        when(apacheDubboProxyService.genericInvoker(null, metaData, exchange)).thenReturn(Mono.empty());
         apacheDubboPlugin = new ApacheDubboPlugin(apacheDubboProxyService);
     }
 
+    @Test(expected = NullPointerException.class)
+    public void doExecute() {
+        SoulContext context = mock(SoulContext.class);
+        exchange.getAttributes().put(Constants.CONTEXT, context);
+        exchange.getAttributes().put(Constants.DUBBO_PARAMS, "{key:value}");
+        exchange.getAttributes().put(Constants.META_DATA, metaData);
+        when(chain.execute(exchange)).thenReturn(Mono.empty());
+        SelectorData selectorData = mock(SelectorData.class);
+        RuleData data = mock(RuleData.class);
+        StepVerifier.create(apacheDubboPlugin.doExecute(exchange, chain, selectorData, data)).expectSubscription().verifyComplete();
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testParameterNotNullExecute() {
+        SoulContext context = mock(SoulContext.class);
+        exchange.getAttributes().put(Constants.CONTEXT, context);
+        metaData.setParameterTypes("parameterTypes");
+        exchange.getAttributes().put(Constants.META_DATA, metaData);
+        SelectorData selectorData = mock(SelectorData.class);
+        RuleData data = mock(RuleData.class);
+        StepVerifier.create(apacheDubboPlugin.doExecute(exchange, chain, selectorData, data)).expectSubscription().verifyComplete();
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testMethodIsNullExecute() {
+        SoulContext context = mock(SoulContext.class);
+        exchange.getAttributes().put(Constants.CONTEXT, context);
+        exchange.getAttributes().put(Constants.DUBBO_PARAMS, "{key:value}");
+        MetaData metaData = MetaData.builder()
+                .id("1332017966661636096")
+                .appName("dubbo")
+                .path("/dubbo/findAll")
+                .serviceName("org.dromara.soul.test.dubbo.api.service.DubboTestService")
+                .rpcType(RpcTypeEnum.DUBBO.getName())
+                .build();
+        exchange.getAttributes().put(Constants.META_DATA, metaData);
+        SelectorData selectorData = mock(SelectorData.class);
+        RuleData data = mock(RuleData.class);
+        StepVerifier.create(apacheDubboPlugin.doExecute(exchange, chain, selectorData, data)).expectSubscription().verifyComplete();
+    }
+
     @Test
-    public void testNamed() {
+    public void named() {
```

Comment:
no need to change name (I think)

Reply:
Ok, I'll change it later. I just deleted what I wrote before with the method name generated by idea tool, so I didn't pay attention to it

Reply:
> no need to change name (I think)

changed